packages/edit-site/src/store/actions.js

```diff
@@ -298,3 +302,136 @@ export function setIsListViewOpened( isOpen ) {
 		isOpen,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	if ( ! isTemplateRevertable( template ) ) {
+		yield controls.dispatch(
+			noticesStore,
+			'createErrorNotice',
+			__( 'This template is not revertable.' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		// This will override the current template entity
+		// so we need to refetch after this call.
+		const fileTemplate = yield controls.resolveSelect(
+			coreStore,
+			'getEntityRecord',
+			'postType',
+			'wp_template',
+			template.id,
+			{ is_custom: false }
+		);
+		if ( ! fileTemplate ) {
+			yield controls.dispatch(
+				noticesStore,
+				'createErrorNotice',
+				__(
+					'The editor has encountered an unexpected error. Please reload.'
+				),
+				{ type: 'snackbar' }
+			);
+			return;
+		}
+		// Refetch the template entity
+		yield controls.resolveSelect(
```

Comment:
I wonder if there is a better way to do this 🤔 

At [line 325](https://github.com/WordPress/gutenberg/pull/28141/files#diff-2df53ac7ebda88cf7debc8c129460578d46471090bceb7f512cd568728a48f73R325) we fetch the file template using `getEntityRecord`. But that overrides the entity of our template. We need to call this again without `is_custom: false` to fetch the real entity (the one that's from the database and not from file).

I tried to use `apiFetch` with `addQueryArgs( template.links.self[0].href, { is_custom: false } )` but that caused problems in specific situations that I can't figure out 🤔

Reply:
> caused problems in specific situations

Are these problems happening during the `apiFetch`, or later in the revert, when handling the file template content?

Behind the scenes, `getEntityRecord` also uses `apiFetch`, sending the result over to `receiveEntityRecords`, which in turns dispatches `receiveItems` which _upserts_ the state.

Throwing a random idea: maybe the problems you encountered were caused by the path?
Have you tried building it in the same way as `getEntityRecord`?

https://github.com/WordPress/gutenberg/blob/4b8b2d9ba2a9b47faf8ecfb54ecebd1acd3f2006/packages/core-data/src/resolvers.js#L113-L116

I think it should turn out to roughly be something like:
```js
const path = addQueryArgs( `/wp/v2/templates/${ template.id }`, {
  is_custom: false,
  context: 'edit',
} );
```